server/server.ts

```diff
@@ -0,0 +1,191 @@
+import bcrypt from 'bcryptjs'
+import { Request, Response, NextFunction } from 'express'
+import jsonServer from 'json-server'
+import jwt from 'jsonwebtoken'
+import { dirname } from 'path'
+import { fileURLToPath } from 'url'
+
+import { DbUser, convertDbUserToUser } from './dbModels'
+import { User, Post, Gender } from '../src/models'
+
+// Type for the database schema
+interface Database {
+  users: DbUser[]
+  posts: Post[]
+}
+
+// Get the directory name of the current module file (for db.json path)
+const __filename = fileURLToPath(import.meta.url)
+const __dirname = dirname(__filename)
+
+// Create a new jsonServer instance
+const server = jsonServer.create()
+
+// Middleware to parse JSON bodies
+server.use(jsonServer.bodyParser)
+
+// Assuming `router.db` is a Lowdb instance with a JSON backend
+const router = jsonServer.router<Database>(`${__dirname}/db.json`)
+// Use default middlewares (for CORS, logging, etc.)
+const middlewares = jsonServer.defaults()
+
+interface RegisterRequestBody {
+  firstName: string
+  lastName: string
+  email: string
+  password: string
+  confirmPassword: string
+  gender: Gender
+}
+
+// Register a custom POST /register endpoint
+server.post('/register', (req: Request, res: Response) => {
+  const {
```

Comment:
We will have validation on the frontend, but, what if i will post from postman? There will be no validation.
To take it a step further, lets add the validation on the backend as well.

We can use https://zod.dev/, as a schema validator, and have a const to will hold the object.
```
const RegisterRequestBodySchema = z.object({
...
```

The same we can do with **LoginRequestBodySchema** 

To parse the request with the validation we can use this method from zod
https://zod.dev/?id=safeparse

This will give us access to 2 new methods,
result.success and results.data ( you must destructure the fields from data)